ts/util/table-builder.ts

```diff
@@ -77,52 +96,69 @@ export class TableBuilder<ParentType, ChildType> {
   }
 
   private addSortControls() {
+    // makes outer wrapper div
     const sortContainer = document.createElement("div");
-    sortContainer.classList.add("sortContainer");
+    sortContainer.classList.add(
+      "sortContainer",
+      "flex",
+      "space-x-2",
+      "items-center",
+      "mt-4"
+    );
     this.container.appendChild(sortContainer);
-
+    // adds sort icon
     const icon = makeIcon("sort");
     icon.title = "Sort";
+    icon.classList.add("text-grey-200");
     sortContainer.appendChild(icon);
 
-    const dropdown = document.createElement("select");
+    // adds all dropdown items
+    let DropdownOptions: BasicDropdownOption[] = [];
+
     this.definition.columns
       .filter((column) => column.sortType)
       .forEach((column) => {
-        this.addSortOption(column, false, dropdown);
-        this.addSortOption(column, true, dropdown);
+        DropdownOptions.push(this.addSortOption(column, false));
+        DropdownOptions.push(this.addSortOption(column, true));
       });
-    dropdown.addEventListener("input", (event) => {
-      const dropdown = <HTMLSelectElement>event.target;
-      const selected = dropdown.options[dropdown.selectedIndex];
-      this.sortColumn = <string>selected.getAttribute("data-column");
-      this.sortDescending = selected.getAttribute("data-order") == "desc";
-      this.reload();
-    });
-    sortContainer.appendChild(dropdown);
+
+    const sortButton = makeDropdownButton();
+    sortButton.innerHTML =
+      this.definition.defaultSort.columnTitle +
+      " - " +
+      this.getSortDescriptor(
+        this.definition.defaultSort.type,
+        this.definition.defaultSort.descending
+      );
+    const dropdownContainer = document.createElement("div");
+    const dropdownMenu = makeDropdownMenu(DropdownOptions, sortButton);
+    const dropdownClickout = makeDropdownClickout();
+
+    const toggleMenu = () => {
+      dropdownMenu.classList.toggle("hidden");
+      dropdownClickout.classList.toggle("hidden");
+    };
+    sortButton.onclick = toggleMenu;
+    dropdownClickout.onclick = toggleMenu;
+
+    dropdownContainer.appendChild(sortButton);
+    dropdownContainer.appendChild(dropdownClickout);
+    dropdownContainer.appendChild(dropdownMenu);
+    sortContainer.appendChild(dropdownContainer);
```

Comment:
This should all be handled by `makeDropdownMenu` in `dropdown.ts`. `makeDropdownMenu` should return the outermost element (`dropdownContainer`) that can be added to the DOM (`sortContainer`). All that should be in the table builder is a call to `makeDropdownMenu` providing the `DropdownOption`s.